docs/data_encryption.md

```diff
@@ -0,0 +1,253 @@
+Encrypting your data
+====================
+
+This example is shown using etcd, but any of the supported backends *should* work.
+
+- [PGP Keypair](#local-testing-keypair)
+  - [Sample Keypair](#local-testing-keypair-sample)
+  - [Generating Your Own Keypair](#local-testing-keypair-generating)
+- [Encrypting & Storing Your Data](#local-testing-saving)
+  - [With GPG](#local-testing-saving-gpg)
+  - [With crypt](#local-testing-saving-crypt)
+- [Verifying Encrypted Data](#local-testing-verify)
+- [Putting It All Together, Using With Confd](#local-testing-confd)
+
+<a name="local-testing-keypair"/>
```

Comment:
What's the rationale behind these internal links? Github automatically generates links like [this](https://github.com/philipsoutham/confd/blob/crypt/docs/data_encryption.md#creating-pgp-keys) on hosted markdown.